### PR TITLE
Moving people component refactor

### PR DESCRIPTION
**Pull request by albrja**
### Refactors components where people move to use vivarium Component sublcasses.
- *Category*: Refactor
- *JIRA issue*: [MIC-4414](https://jira.ihme.washington.edu/browse/MIC-4414)
- *Research reference*: 

### Changes and notes
-refactors person and household immigration/emigration to use vivarium component

### Verification and Testing
Successfully ran simulation

### CODE REVIEW

File: src/vivarium_census_prl_synth_pop/components/immigration.py [hunk 1]
```diff
@@ -13,21 +14,22 @@
...
-
     ##############
     # Properties #
     ##############
 
     @property
-    def name(self):
-        return "immigration"
+    def columns_required(self) -> List[str]:
+        return []
```
Comment (rmudambi) [inline src/vivarium_census_prl_synth_pop/components/immigration.py:28]: Does this component needs all columns in its population view? I don't see a population view at all in the old code.

File: src/vivarium_census_prl_synth_pop/components/person_migration.py [hunk 2]
```diff
@@ -24,16 +26,27 @@ class PersonMigration:
...
+        return [
+            "household_id",
+            "relationship_to_reference_person",
+            "previous_timestep_address_id",
+        ]
+
+    @property
+    def initialization_requires(self) -> Dict[str, List[str]]:
+        return {
+            "requires_columns": ["previous_timestep_address_id"],
```
Comment (rmudambi) [inline src/vivarium_census_prl_synth_pop/components/person_migration.py:44]: In the previous version it looks like this component creates the column `previous_timestep_address_id` but here you're saying it requires it in initialization.

**Review by rmudambi — CHANGES_REQUESTED** [review]

File: src/vivarium_census_prl_synth_pop/components/person_migration.py (hunk 2 above)
Comment (albrja) [reply src/vivarium_census_prl_synth_pop/components/person_migration.py:44]: Yes this was a mistake.

File: src/vivarium_census_prl_synth_pop/components/immigration.py (hunk 1 above)
Comment (albrja) [reply src/vivarium_census_prl_synth_pop/components/immigration.py]: I was unsure of this. There is no population view but doesn't the component still need to use columns?

**Review by rmudambi — APPROVED** [review]

**Pull request merged**